handlers/feed.go

```diff
@@ -61,84 +68,92 @@ func DownloadYoutubeFeed(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	var dataCount int64 = 0
 	for i := 0; i < len(youtube_download.YoutubeUrls); i++ {
 		url := youtube_download.YoutubeUrls[i]
-
-		feed, err := feeds.ParseFeed(url, false)
-		if err == nil {
-			processYoutubeDownload(url, *feed)
+		// Split URL to get video ID
+		idSplit := strings.Split(url, "v=")
+		id := idSplit[1]
+
+		// Make a Youtube API call to check if the video exists
+		call := tube.Videos.List([]string{"snippet,contentDetails,statistics"}).Id(id)
+		response, _ := call.Do()
+
+		// Add the Youtube results to the data result it should be one if the video exists
+		if response.PageInfo.TotalResults < 1 {
+			break
+		} else {
+			dataCount += response.PageInfo.TotalResults
 		}
```

Comment:
we dont need to increment data count if we break then datacount < youtube_download.YoutubeUrls so just put this there and then get rid of the if statement after
```
w.WriteHeader(http.StatusBadRequest)
		json.NewEncoder(w).Encode("Could not process Youtube download, one of the youtueb videos does not exists")
```